mcs/class/Mono.Security/Mono.Security.X509/PKCS12.cs

```diff
@@ -741,22 +741,26 @@ private DSAParameters GetExistingParameters (out bool found)
 		private void AddPrivateKey (PKCS8.PrivateKeyInfo pki) 
 		{
 			byte[] privateKey = pki.PrivateKey;
-			switch (privateKey [0]) {
-				case 0x02:
+			try {
+				switch (pki.Algorithm) {
+				case "1.2.840.113549.1.1.1": // Oids.RsaRsa
```

Comment:
Would be better to use existing constants from https://github.com/mono/mono/blob/master/mcs/class/Mono.Security/Mono.Security.X509/X509Certificate.cs#L73 instead of comments (just make them internal)